CMakeLists.txt

```diff
@@ -1,19 +1,53 @@
-cmake_minimum_required(VERSION 2.8)
-project(mustache)
-if (UNIX)
-	add_definitions(
-		-Wall
-		-Wextra
-		-Werror
-		-std=c++11
-	)
-elseif (MSVC)
-	add_definitions(
-		/W3
-		/WX
-	)
+cmake_minimum_required(VERSION 3.7.2)
+
+project(mustache VERSION 4.0.0 LANGUAGES CXX)
+
+option(MUSTACHE_BUILD_TESTS
+       "build the tests with catch"
+       ON)
+
+add_library(${PROJECT_NAME}
+            INTERFACE)
+target_include_directories(${PROJECT_NAME}
+                           INTERFACE $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/>
+                                     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+
+if(MUSTACHE_BUILD_TESTS)
+  if(UNIX)
+    add_definitions(-Wall
```

Comment:
If we are moving to CMake 3.1 or later, let's use the modern target_* functions instead of the global functions.